### DISCUSSION TITLE
Form in Form works although messy. Question.

### DISCUSSION THREAD

**Discussion by gmentzik** [Budibase/budibase#8803] [Help] [answered]
Although according to documentation Form in form is not allowed, I have successfully created multistep forms that have a form in each page by creating a form outside and dragging it inside the parent Form afterwards. At actions Budibase goes nuts showing fields of internal forms as belonging to externals but in reality form fields only work (are addressable and readable) only when their immediate parent Form is set. In order to avoid conflict issues when allowing Budibase to handle save row by using form though you may need to make sure there are not same named fields in outer and inner forms or hanldle save manually (specify which fields to use yourself).  So you can save a row or handle a field making sure you refer to correct form. So besides that messy representation that gets worst the more forms you add inside the main is there another reason you suggest to not use Form in Form?

**Comment by insomniacj93** [comment, marked as answer]
@gmentzik from self experience this is not reliable, one change in either form and you can get everything tangled up.
I suggest you to rely more on form steps instead, there's no need to have a form inside a form. The step forms will perform accordingly to expectations, you can also navigate between steps as you wish by using navigate to specific form step. You can create really dinamic forms with that. Either way, if you managed to make it work coolio! I didn't ^^' .

**Reply by gmentzik** [reply]
@insomniacj93 
The way the guide suggest to create a multi step form is ok for New entry. But what happens if you want to edit using a multistep form. To do that you have to fill all sub form fields manually, loosing that nice autofill feature that update form provides :( Maybe if field group could be connected to a table schema would be a solution to such issues.

**Reply by melohagan** [reply]
Hey @gmentzik The autofill on multi-step edit form is working for me:

![Screenshot 2022-12-05 at 15 19 40](https://user-images.githubusercontent.com/101575380/205673919-f539324a-a1a2-4b37-90f3-5a2a536dcb27.png)

![Screenshot 2022-12-05 at 15 19 56](https://user-images.githubusercontent.com/101575380/205673990-9732a830-920f-4dac-a610-fe286176e653.png)

![Screenshot 2022-12-05 at 15 20 19](https://user-images.githubusercontent.com/101575380/205674073-160d0ea3-28d9-44cf-96ca-20f3949d0426.png)

**Reply by gmentzik** [reply]
Yes but what about step 2. Will it load the adress data from Address table ?
2ndly I have a case where depending on a value of a field of 1st (parent) form , I display/enable different sub forms that relate to different tables and need to filled in. More over what will happen if field names are of different forms are same? Using form in form I was able to easily bring back the data in edit case of any of these forms adding to each one the related provider-> repeater and there is no conflict even in case of same field names. Alternatively I would have to load these fields one by one as form will load the data from the directly related provider->repeater. I am not sure what would happen if I have relations created between tables, since I am using external db and handling relations in the db schema mostly. I have successfully created also a form the way you describe in your guide but generally as the guide suggests is more to split a large table into a form of several pages.
I would suggest that we should be able to at least insert one form under each step, and Budibase should display only the binding fields that are directly under a form , not those under a subform. Then it would be very clear on which form you are working with. I am still not aware of the design difficulties you may face for that so I guess that the precautionary measure you have taken to treat form as top level component as you mention is something you feel is the safest approach.
Alternatively I would suggest as future improvement to create a multistep wizard component that would not depend on a form and allow us to put whatever we want inside each step because frankly this is that I thought multi step form was able to do but it seams not. Of course this wizard should allow to save the multiple inlcuded forms at the final or intermediate steps depending on needed implementation. 
Anyway thank you for support. I will try to redesign my multi step forms to not have future issues. Also please update your guide here : https://docs.budibase.com/docs/form-steps to make clear that Form in Form is not allowed or not advised. Whatever you feel more appropriate.

**Comment by gmentzik** [comment]
@melohagan 
So I have the following scenario.
I have Provider1 -> Repeater1 -> Provider2 -> Form with data from Provider 2.
I want to filter all provider 2 in repeater by a specific field value. To Do that I create the structure as follows: 
Filter Form for Provider2 (field1 (picker)) -> Provider1 -> Repeater1 -> Provider2 -> Form with data from Provider 2 (field2 (text))

Budibase allows me to do that and it works fine, beside the fact that when at filter of Provider 2 at selecting field from Filter Form it gives the option to select field1 and field2 which does not work even if you select it and of course is wrong as field2 does not belong to Filter Form (this is the mistake I was mentioning). 
So why in this case is allowed? What is different? And if this not a correct approach how can I do a dynamic filtering in this case if form in form is not allowed but you cannot read / update field unless you are inside a form?
